text/0000-int-guidelines.md

```diff
@@ -0,0 +1,164 @@
+- Start Date: 2015-01-26
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+
+This RFC proposes usage guidelines for the various integer types.
+
+# Motivation
+
+The goal of this RFC is to help people decide what integer types to use when they need
+to make a decision for a new API.
+
+It builds on [https://github.com/rust-lang/rfcs/pull/560](the integer overflow RFC),
+which provides debug-time assertions for overflow and underflow. One of the goals
+of this RFC is to provide guidance for when to use unsigned types, and these
+assertions affect the traditional tradeoffs about unsigned types, which are described
+under the detailed design below.
+
+It also draws inspiration from the [Google C++ Style Guide](http://google-styleguide.googlecode.com/svn/trunk/cppguide.html#Integer_Types)
```

Comment:
> The Google C++ Style Guide is based on the premise that having a very small language is more valuable than anything else because it makes the code accessible to people who only know Java. Most of the recommendations in it are solely based on that school of thought, which is diametrically opposed to Rust's language design. Their reasoning for forbidding unsigned integers has a lot more to do with removing choice and aligning C++ with Java than anything else.

This is nonsense, sorry. I read the (constant) internal discussions that precede changes to the Google C++ Style guide, and what you've said above is incorrect. The reasoning is always "what can we do here to prevent nasty bugs" and never has anything to do with Java.